modules/deployment/src/test/java/org/mule/runtime/module/deployment/internal/DeploymentServiceTestCase.java

```diff
@@ -4205,12 +4123,43 @@ private void executeApplicationFlow(String flowName) throws MuleException {
         .build());
   }
 
-  private void assertZombieApplication(String appId) {
-    final Map<URI, Long> zombieMap = deploymentService.getZombieApplications();
-    assertEquals("Wrong number of zombie apps registered.", 1, zombieMap.size());
-    final Map.Entry<URI, Long> zombie = zombieMap.entrySet().iterator().next();
-    assertEquals("Wrong URL tagged as zombie.", appId, new File(zombie.getKey()).getParentFile().getName());
-    assertTrue("Invalid lastModified value for file URL.", zombie.getValue() != -1);
+  private void assertZombieApplication(String appId, int expectedZombieCount, boolean parent) {
+    final Map<String, Map<URI, Long>> zombieMap = deploymentService.getZombieApplications();
+    assertZombie(zombieMap, appId, expectedZombieCount, parent);
+  }
+
+  private void assertZombieDomain(String domainId, int expectedZombieCount, boolean parent, boolean expectName) {
+    final Map<String, Map<URI, Long>> zombieMap = deploymentService.getZombieDomains();
+    assertZombie(zombieMap, domainId, expectedZombieCount, parent);
+  }
+
+  private void assertZombie(Map<String, Map<URI, Long>> zombieMap, String appId, int expectedZombieCount, boolean parent) {
+    assertEquals("Wrong number of zombie apps registered.", expectedZombieCount, zombieMap.size());
+    if (expectedZombieCount > 0) {
+      if (parent) {
+        assertThat("Wrong URL tagged as zombie.",
+                   getZombieFromMap((entry) -> new File((URI) entry.getKey()).getParentFile().getName().equals(appId), zombieMap),
+                   is(notNullValue()));
+      } else {
+        assertThat("Wrong URL tagged as zombie.",
+                   getZombieFromMap((entry) -> new File((URI) entry.getKey()).getName().equals(appId), zombieMap),
+                   is(notNullValue()));
+      }
+      assertThat("Invalid lastModified value for file URL.", zombieMap.get(appId), not(hasValue(-1)));
+    }
+  }
+
+  private Map.Entry<URI, Long> getZombieFromMap(Predicate<Map.Entry> filter, Map<String, Map<URI, Long>> zombieMap) {
+    Map.Entry<URI, Long> zombieEntry = null;
+    //For every app in zombieMap
+    for (Map<URI, Long> zombieResource : zombieMap.values()) {
+      try {
+        zombieEntry = zombieResource.entrySet().stream().filter(filter).collect(Collectors.toList()).get(0);
```

Comment:
Use findFirst and check if there is a value instead of relying on an exception being thrown